src/v/security/audit/schemas/tests/ocsf_schemas_test.cc

```diff
@@ -1066,3 +1066,45 @@ BOOST_AUTO_TEST_CASE(make_authn_event_failure) {
     auto ser = sa::rjson_serialize(authn);
     BOOST_REQUIRE_EQUAL(ser, ::json::minify(expected));
 }
+
+BOOST_AUTO_TEST_CASE(test_ocsf_size) {
```

Comment:
can we add a unit test for each type so that it triggers if someone adds something new to a structure in a way that is incompatible?

Reply:
So if someone adds a new OCSF struct to an OCSF message and it doesn't have an `equality_fields()` method, this will be a compilation error. Is that along the lines of what you were thinking?

Reply:
presumably

Reply:
Yeah thats also why i didn't make this a free floating method so you couldn't just pass any T to it.